activerecord/test/cases/associations/has_and_belongs_to_many_associations_test.rb

```diff
@@ -168,6 +168,21 @@ def test_proper_usage_of_primary_keys_and_join_table
     assert_equal 1, country.treaties.count
   end
 
+  def test_join_table_ignores_primary_key_prefix_type
+    old_primary_key_prefix_type = ActiveRecord::Base.primary_key_prefix_type
+
+    ActiveRecord::Base.primary_key_prefix_type = :table_name_with_underscore
+
+    load 'models/country.rb'
```

Comment:
I need to reset the join model's primary key definition, but I want to avoid directly referencing the model as it's a private constant. 

The more direct way to do it is
```ruby
Country.const_get(:HABTM_Treaties).reset_primary_key
```

I'm not a fan of either approach, so open to suggestions